foundation-extensions/mobile-core/platform-services.md

```diff
@@ -1,6 +1,122 @@
+
+The following code snippet details how to make a simple network request and handle the response.
+
+{% tabs %}
+{% tab title="Android" %}
+
+```java
+AndroidNetworkService androidNetworkService = new AndroidNetworkService(ServiceProvider.getInstance().getNetworkService());
+androidNetworkService.connectUrlAsync(url,
+											  POST, payload,
+		null, 10000, 10000, new NetworkService.Callback() {
+			@Override
+			public void call(NetworkService.HttpConnection connection) {
+				// handle `httpConnection`
+			}
+});
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Create your `NetworkRequest`, for more details see `NetworkRequest.swift`
+let networkRequest = NetworkRequest(url: url, httpMethod: .get, httpHeaders: headers)
+
+// Get an instance of the current network service
+let networkService = ServiceProvider.shared.networkService
+
+// Make a request
+networkService.connectAsync(networkRequest: networkRequest) { httpConnection in
+  // handle `httpConnection`
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+### DataQueuing
+
+The `DataQueuing` service provides access to FIFO queues. This service is particularly useful when used in conjunction with a `PersistentHitQueue`.
+
+The following code snippet shows how to create a `DataQueue` and add a `DataEntity` to the queue.
+
+{% tabs %}
+{% tab title="Android" %}
+
+```java
+DataQueue dataQueue = ServiceProvider.getInstance().getDataQueueService().getDataQueue(name);
+DataEntity dataEntity = new DataEntity(mydata);
+dataQueue.add(dataEntity);
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Create a `DataQueue`
+guard let dataQueue = ServiceProvider.shared.dataQueueService.getDataQueue(label: name) else {
+  Log.error(label: "\(name):\(#function)", "Failed to create Data Queue")
+  return
+}
+
+// Create a `DataEntity`
+let entity = DataEntity(data: myData)
+
+// Add entity to `dataQueue`
+dataQueue.add(entity)
+```
+
+{% endtab %}
+{% endtabs %}
+
+### SystemInfoService (iOS) & DeviceInforming (Android)
+
+The `SystemInfoService` (iOS) and the `DeviceInforming`(Android) service let you access critical pieces of information related to the user's device, such as carrier name, device name, locale, and more. 
+
+The following code snippet shows how to invoke the API to retrieve the user's active locale.
+
+{% tabs %}
+{% tab title="Android" %}
+
+```java
+DeviceInforming deviceInfoService = ServiceProvider.getInstance().getDeviceInfoService();
+String localName = deviceInfoService.getLocaleString();
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Add a computed variable to your type or use it direclty in the function where required
+private var systemInfoService: SystemInfoService {
+  return ServiceProvider.shared.systemInfoService
+}
+
+// ...
+let locale = systemInfoService.getActiveLocaleName()
+```
+
+{% endtab %}
+{% endtabs %}
```

Comment:
Wondering if we need to list out all of the APIs exposed by the service? With snippets for each.